server/raft.go

```diff
@@ -3121,9 +3121,13 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 	if n.State() == Leader {
 		// If we are the same we should step down to break the tie.
 		if ae.term >= n.term {
-			n.term = ae.term
-			n.vote = noVote
-			n.writeTermVote()
+			// If it's the same term, don't clear our vote, otherwise we could
```

Comment:
Comment references old behavior. Could be clear to just talk about new behavior, eg:
 > If term is higher than previously seen, clear vote

The implicit `else` case is also interesting here.
 - This node is leader
 - Some other node is sending an append entry with the same term

This should never ever happen. Each term can have at most one leader (if everyone is complying to the rules)

Reply:
Fixed the comment at least.